packages/application-shell-connectors/src/components/application-context/normalizers.ts

```diff
@@ -1,9 +1,46 @@
 import type {
-  Maybe,
-  TFetchProjectQuery,
+  TAppliedPermission,
+  TAppliedActionRight,
+  TAppliedMenuVisibilities,
   TStoreDataFence,
 } from '../../types/generated/mc';
 
+// Menu visibilities
+export type TMenuVisibilities = {
```

Comment:
It seems to be a matter of subjective opinion but I like `Record<string, boolean>` for these type of things. Keep it as is if you want to :)